corehq/apps/campaign/views.py

```diff
@@ -1,22 +1,165 @@
+from corehq.apps.domain.views.base import BaseDomainView
+from corehq.apps.es.case_search import CaseSearchES, case_property_missing
+from corehq.apps.geospatial.utils import get_lat_lon_from_dict
+from corehq.apps.hqwebapp.crispy import CSS_ACTION_CLASS
 from corehq.apps.hqwebapp.decorators import use_bootstrap5
+from corehq.apps.reports.generic import get_filter_classes
+from corehq.apps.reports.standard.cases.basic import CaseListMixin
 from corehq.apps.reports.views import BaseProjectReportSectionView
+from corehq.form_processor.models import CommCareCase
+from corehq.util.timezones.utils import get_timezone
+
+
+class DashboardMapFilterMixin(object):
+    fields = [
+        'corehq.apps.reports.filters.case_list.CaseListFilter',
+        'corehq.apps.reports.standard.cases.filters.CaseSearchFilter',
+    ]
+
+    def dashboard_map_case_filters_context(self):
+        return {
+            'report': {
+                'title': self.page_title,
+                'section_name': self.section_name,
+                'show_filters': True,
+            },
+            'report_filters': [
+                dict(field=f.render(), slug=f.slug) for f in self.dashboard_map_filter_classes
+            ],
+            'report_filter_form_action_css_class': CSS_ACTION_CLASS,
+        }
+
+    @property
+    @memoized
+    def dashboard_map_filter_classes(self):
+        timezone = get_timezone(self.request, self.domain)
+        return get_filter_classes(self.fields, self.request, self.domain, timezone)
 
 
 @method_decorator(login_and_domain_required, name='dispatch')
 @method_decorator(toggles.CAMPAIGN_DASHBOARD.required_decorator(), name='dispatch')
 @method_decorator(use_bootstrap5, name='dispatch')
-class DashboardView(BaseProjectReportSectionView):
+class DashboardView(BaseProjectReportSectionView, DashboardMapFilterMixin):
     urlname = 'campaign_dashboard'
     page_title = gettext_lazy("Campaign Dashboard")
     template_name = 'campaign/dashboard.html'
 
     @property
     def page_url(self):
         return reverse(self.urlname, args=[self.domain])
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({
+            'mapbox_access_token': settings.MAPBOX_ACCESS_TOKEN,
+            'map_widgets': self._dashboard_map_configs,
+        })
+        context.update(self.dashboard_map_case_filters_context())
+        return context
+
+    @property
+    def _dashboard_map_configs(self):
+        dashboard_maps = Dashboard.objects.get(domain=self.domain).maps.all()
+        dashboard_map_configs = {
+            'cases': [],
+            'mobile_workers': [],
+        }
+        for dashboard_map in dashboard_maps:
+            config = {
+                'id': dashboard_map.id,
+                'title': dashboard_map.title,
+                'case_type': dashboard_map.case_type,
+                'gps_prop_name': dashboard_map.geo_case_property,
+            }
```

Comment:
You could use `model_to_dict` for this
```python
from django.forms.models import model_to_dict
```
```suggestion
            config = model_to_dict(dashboard_map)
```
but you would need to rename "gps_prop_name" to "geo_case_property" in the front-end code.